src/report/business-reports/taxinvoice.eguile.scm

```diff
@@ -82,18 +82,30 @@
                          (< (car (gnc-transaction-get-date-posted t1))
                             (car (gnc-transaction-get-date-posted t2))))))))
 
+
+      ;; Is this an invoice or something else
+      (if (not (null? opt-invoice))
+          (begin
+            (set! owner (gncInvoiceGetOwner  opt-invoice))
+            (let ((type (gncInvoiceGetType  opt-invoice)))
+              (cond
+               ((eqv? type GNC-INVOICE-CUST-INVOICE)
+                (set! cust-doc? #t))
+               
+               ))))
+      
       ; pre-scan invoice entries to look for discounts and taxes
       (for entry in entries do
           (let ((action    (gncEntryGetAction entry))
                 (qty       (gncEntryGetDocQuantity entry credit-note?))
                 (discount  (gncEntryGetInvDiscount entry))
-                (taxable?   (gncEntryGetInvTaxable entry))
-                (taxtable  (gncEntryGetInvTaxTable entry)))
+                (taxable?  (if cust-doc? (gncEntryGetInvTaxable entry) (gncEntryGetBillTaxable entry)))
+                (taxtable  (if cust-doc? (gncEntryGetInvTaxTable entry) (gncEntryGetBillTaxTable entry))))
             (if (not (string=? action ""))
               (set! units? #t))
             (if (not (= (gnc-numeric-to-double qty) 1.0))
               (set! qty? #t))
-            (if (not (gnc-numeric-zero-p discount)) (set! discount? #t))
+            (if (not (or cust-doc? (gnc-numeric-zero-p discount))) (set! discount? #t))
             ;(if taxable - no, this flag is redundant
```

Comment:
This looks like a logic error to me ? Assuming discounts are only possible for customer documents the condition is:

discount? is true if we're handling a customer document and and the discount value is non zero

Or translated into scheme

`(and cust-doc? (not (gnc-numeric-zero-p discount)))`

If you want to switch this to an or form it would become

`(not (or (not cust-doc?) (gnc-numeric-zero-p discount)))`

Feel free to pick your preferred format.